src/nfl_managers.py

```diff
@@ -37,207 +23,83 @@ class BaseNFLManager: # Renamed class
             if cached_data:
                 # Validate cached data structure
                 if isinstance(cached_data, dict) and 'events' in cached_data:
-                    self.logger.info(f"[NFL] Using cached schedule for {current_year}")
+                    self.logger.info(f"[NFL] Using cached schedule for {season_year}")
                     return cached_data
                 elif isinstance(cached_data, list):
                     # Handle old cache format (list of events)
-                    self.logger.info(f"[NFL] Using cached schedule for {current_year} (legacy format)")
+                    self.logger.info(f"[NFL] Using cached schedule for {season_year} (legacy format)")
                     return {'events': cached_data}
                 else:
-                    self.logger.warning(f"[NFL] Invalid cached data format for {current_year}: {type(cached_data)}")
+                    self.logger.warning(f"[NFL] Invalid cached data format for {season_year}: {type(cached_data)}")
                     # Clear invalid cache
                     self.cache_manager.delete(cache_key)
         
         # If background service is disabled, fall back to synchronous fetch
         if not self.background_enabled or not self.background_service:
             return self._fetch_nfl_api_data_sync(use_cache)
         
-        # Check if we already have a background fetch in progress
-        if current_year in self.background_fetch_requests:
-            request_id = self.background_fetch_requests[current_year]
-            result = self.background_service.get_result(request_id)
-            
-            if result and result.success:
-                self.logger.info(f"[NFL] Background fetch completed for {current_year}")
-                # Validate result data structure
-                if isinstance(result.data, dict) and 'events' in result.data:
-                    return result.data
-                elif isinstance(result.data, list):
-                    # Handle case where result.data is just the events list
-                    return {'events': result.data}
-                else:
-                    self.logger.error(f"[NFL] Invalid background fetch result format: {type(result.data)}")
-                    return None
-            elif result and not result.success:
-                self.logger.warning(f"[NFL] Background fetch failed for {current_year}: {result.error}")
-                # Remove failed request and try again
-                del self.background_fetch_requests[current_year]
-            else:
-                self.logger.info(f"[NFL] Background fetch in progress for {current_year}, using partial data")
-                # Return partial data if available, or None to indicate no data yet
-                partial_data = self._get_partial_nfl_data(current_year)
-                if partial_data:
-                    return {'events': partial_data}
-                return None
```

Comment:
This is related to #47. though i haven't tested anything since #50 was merged.

Reply:
I consolidated the polling even more, so Sports Core now has a function to pull events for the week (partial update while background is running) and that day (mainly for live)

Reply:
without it and just using the background and todays pulls, i'm getting super good updates. Even rotating between NCAA FB and MLB